astropy/wcs/src/str_list_proxy.c

```diff
@@ -210,52 +214,33 @@ static PySequenceMethods PyStrListProxy_sequence_methods = {
   NULL
 };
 
-static PyTypeObject PyStrListProxyType = {
-  PyVarObject_HEAD_INIT(NULL, 0)
-  "astropy.wcs.StrListProxy", /*tp_name*/
-  sizeof(PyStrListProxy),  /*tp_basicsize*/
-  0,                          /*tp_itemsize*/
-  (destructor)PyStrListProxy_dealloc, /*tp_dealloc*/
-  0,                          /*tp_print*/
-  0,                          /*tp_getattr*/
-  0,                          /*tp_setattr*/
-  0,                          /*tp_compare*/
-  (reprfunc)PyStrListProxy_repr, /*tp_repr*/
-  0,                          /*tp_as_number*/
-  &PyStrListProxy_sequence_methods, /*tp_as_sequence*/
-  0,                          /*tp_as_mapping*/
-  0,                          /*tp_hash */
-  0,                          /*tp_call*/
-  (reprfunc)PyStrListProxy_repr, /*tp_str*/
-  0,                          /*tp_getattro*/
-  0,                          /*tp_setattro*/
-  0,                          /*tp_as_buffer*/
-  Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_GC, /*tp_flags*/
-  0,                          /* tp_doc */
-  (traverseproc)PyStrListProxy_traverse, /* tp_traverse */
-  (inquiry)PyStrListProxy_clear, /* tp_clear */
-  0,                          /* tp_richcompare */
-  0,                          /* tp_weaklistoffset */
-  0,                          /* tp_iter */
-  0,                          /* tp_iternext */
-  0,                          /* tp_methods */
-  0,                          /* tp_members */
-  0,                          /* tp_getset */
-  0,                          /* tp_base */
-  0,                          /* tp_dict */
-  0,                          /* tp_descr_get */
-  0,                          /* tp_descr_set */
-  0,                          /* tp_dictoffset */
-  0,                          /* tp_init */
-  0,                          /* tp_alloc */
-  PyStrListProxy_new,      /* tp_new */
+static PyType_Slot PyStrListProxyType_slots[] = {
+  {Py_tp_dealloc, (destructor)PyStrListProxy_dealloc},
+  {Py_tp_repr, (reprfunc)PyStrListProxy_repr},
+  {Py_tp_as_sequence, &PyStrListProxy_sequence_methods}, // FIXME: this Py_tp_* doesn't exist
```

Comment:
The slots are `Py_sq_length`, `Py_sq_item` and `Py_sq_ass_item` rather than the whole `PySequenceMethods` struct

Reply:
```suggestion
  {Py_sq_length, (void*)PyStrListProxy_len},
  {Py_sq_item, (void*)PyStrListProxy_getitem},
  {Py_sq_ass_item, (void*)PyStrListProxy_setitem},
```

+ delete the static PySequenceMethods struct above